### ISSUE TITLE
Binary scenes don't save instanced scenes in them

### ISSUE THREAD

**Issue by Summersay415** [godotengine/godot#80783]
### Godot version

3.6.beta3

### System information

Ubuntu Studio 23.04

### Issue description

When saving binary scenes, references to instanced scenes are not saved. So, reopening saved scene causes loss of references to instanced scenes (only topmost node is saved, without reference). This bug is 3.x only, 3.5.2 don't have it.
Caused by: https://github.com/godotengine/godot/pull/62408

### Steps to reproduce

1. Create scene.
2. Instance another scene inside.
3. Save first scene with .scn (binary) extension.
4. Reopen it. Reference to instanced scene is lost.

https://github.com/godotengine/godot/assets/71974498/666ad675-c1da-45fe-836b-8800d9383a0a

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
Can you reproduce this in 3.6.beta1 and 3.6.beta2?

**Comment by Summersay415** [comment]
> Can you reproduce this in 3.6.beta1 and 3.6.beta2?

Yes

**Comment by Calinou** [comment]
If you can [compile the engine from source](https://docs.godotengine.org/en/stable/development/compiling/index.html), you could look into [bisecting the regression](https://docs.godotengine.org/en/stable/community/contributing/bisecting_regressions.html) to greatly speed up troubleshooting.

Use these commands to start bisecting:

```bash
git bisect start
git bisect good 3.5.2-stable

# This is 3.6.beta1's commit hash.
git bisect bad 632a544c6e8f847d6796846d44f01231d1744958
```

**Comment by Summersay415** [comment]
I will try it in free time!

**Comment by akien-mga** [comment]
> `git bisect good 3.5.2-stable`

I would advise starting from `3.5-stable` instead, bisecting two diverging branches will add more steps unnecessarily.

**Comment by Summersay415** [comment]
Bisected:
2ceb93bbeff6f5c7b6eed4ea79aff5f0574f4f92 is the first bad commit
commit 2ceb93bbeff6f5c7b6eed4ea79aff5f0574f4f92
Author: Francois Belair <razoric480@gmail.com>
Date:   Sat Jun 25 15:51:37 2022 -0400

    Fix custom res caching sub-res even if no-cache
    
    Fixes #59686, fixes #59752

**Comment by akien-mga** [comment]
CC @Razoric480 @KoBeWi

**Comment by Razoric480** [comment]
Can confirm that it is _saved_ in the .scn file (if the saving happens in 3.x but the loading in 3.5-stable, the reference to the external TSCN is maintained.)

**Comment by akien-mga** [comment]
Fixed by #81026.
